kirovy/models/cnc_user.py

```diff
@@ -0,0 +1,65 @@
+import uuid
+
+from django.contrib.auth.models import AbstractBaseUser
+from django.db import models
+from django.utils.translation import gettext as _
+from kirovy import typing as t
+
+from kirovy.objects import CncnetUserInfo
+
+__all__ = ["CncUser"]
+
+
+class CncUser(AbstractBaseUser):
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    cnc_net_id = models.IntegerField(
+        unique=True,
+        editable=False,
+        help_text=_("The user ID from the CNCNet ladder API."),
+    )
+
+    username = models.CharField(
+        null=True, help_text=_("The name from the CNCNet ladder API.")
+    )
+    """:attr: The username for debugging purposes. Don't rely on this field for much else."""
+
+    verified_map_uploader = models.BooleanField(null=False, default=False)
+    """:attr: This user is a map uploader verified by cncnet staff."""
+
+    verified_email = models.BooleanField(null=False, default=False)
+
+    group = models.CharField(
+        null=False, help_text=_("The user group from the CNCNet ladder API.")
+    )
+
+    USERNAME_FIELD = "cnc_net_id"
+
+    def save(self, *args, **kwargs):
+        self.set_unusable_password()
+        super().save(*args, **kwargs)
+
+    @property
+    def can_upload(self) -> bool:
+        return self.verified_map_uploader or self.verified_email
+
+    @staticmethod
+    def create_or_update_from_cnc_net(user_dto: CncnetUserInfo) -> "CncUser":
```

Comment:
ditto

Reply:
Also inconsistent `Cncnet` vs `CncNet`, not sure about the naming violations regarding acronyms in PascalCase but I would either pick `Cncnet` or `CnCNet`.

Reply:
Yeah I'll switch all of these. Embarrassing how inconsistent I was.

Reply:
No worries haha